overlord/configstate/configcore/network.go

```diff
@@ -0,0 +1,86 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+)
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.network.disable-ipv6"] = true
+}
+
+func validateNetworkSettings(tr Conf) error {
+	return validateBoolFlag(tr, "network.disable-ipv6")
+}
+
+func handleNetworkConfiguration(tr Conf) error {
+	dir := filepath.Join(dirs.GlobalRootDir, "/etc/sysctl.d")
+	name := "10-snapd-network.conf"
+	content := bytes.NewBuffer(nil)
+
+	output, err := coreCfg(tr, "network.disable-ipv6")
+	if err != nil {
+		return nil
+	}
+	switch output {
+	case "true":
+		sysctl := "net.ipv6.conf.all.disable_ipv6=1"
+		content.WriteString(sysctl + "\n")
+	case "false":
+		sysctl := "net.ipv6.conf.all.disable_ipv6=0"
+		content.WriteString(sysctl + "\n")
+	case "":
+		// nothing
+	default:
+		return fmt.Errorf("unsupported disable-ipv6 option: %q", output)
+	}
+	dirContent := map[string]*osutil.FileState{}
+	if content.Len() > 0 {
+		dirContent[name] = &osutil.FileState{
+			Content: content.Bytes(),
+			Mode:    0644,
+		}
+	}
+
+	// write the new config
+	glob := name
+	changed, _, err := osutil.EnsureDirState(dir, glob, dirContent)
```

Comment:
The removals are a bit tricky - right now the removal of the sysctl file will just mean the system remains in the state it was in before. To fix this we essentially need to do the following:
1. if disable-ipv6 is set we create the sysctl file and load it
2. if disable-ipv6 is false or empty we remove the sysctl file and manually do `sysctl -w net.ipv6.conf.all.disable-ipv6=1`

Reply:
When you don't generate any content and thus the file gets removed (by us) then this would trigger sysctl call below. I didn't mean if someone comes along and removes the file behind our back.

Alternatively we could implement a new helper like SyncDir but for sysctl options. Then we would use both. One (this one) for on-boot behavior and other (new one) for the in-kernel value. The file would not trigger any changes, instead all of the changes would be done by ensure-sysctl.